Automattic/wp-calypso
# Plans: move HappinessSupport option to a card and style the new header.

Part of solution to #22691, as per new designs for My Plan tab for both Jetpack and .com plans.

Main changes included:
- extract `HappinessSupport` components from the header into a separate feature card. Note: it is not wrapped in PurchaseDetail component as the remaining feature cards, because it is not stateless and has much more complex logic (a bit of a peculiar snowflake among those),
- adjust the header to spread over the entire viewport and use flexbox to align its elements.

**To test:**
- checkout this branch in your local Calypso,
- go to `http://calypso.localhost:3000/plans/my-plan/:site` for Jetpack and WP sites on different plans,
- verify that the plan is correctly detected and HappinessSupport card has appropriate header (Free vs. paid plan). The visuals should look as below,
- verify that for .com Business and Jetpack Pro plans the `Ask a question` button triggers HappyChat (live when available), while for other plans a redirect to `https://jetpack.com/contact-support/?rel=support`.
- run the unit tests for the `ProductPurchaseFeaturesList` component

Note: .com plans have no My Plan tab for the Free plan.

.com plans:

![screen shot 2018-03-12 at 13 28 38](https://user-images.githubusercontent.com/13561163/37286107-ef9dcd2e-25f8-11e8-9bea-2fdf6f6e80ce.png)

Jetpack plans:

FREE
![screen shot 2018-03-12 at 18 03 37](https://user-images.githubusercontent.com/13561163/37301133-a47575fa-261f-11e8-9b2c-c048370db179.png)

PAID
![screen shot 2018-03-12 at 13 53 36](https://user-images.githubusercontent.com/13561163/37287304-718047ec-25fc-11e8-9464-46583c80094a.png)



Up next:
- align feature cards to form symmetric grid and set unified size,
- update copy and icon positioning.
